ext/web/02_event.js

```diff
@@ -736,12 +736,8 @@ function innerInvokeEventListeners(
     return found;
   }
 
-  let handlers = targetListeners[type];
-
   // Copy event listeners before iterating since the list can be modified during the iteration.
-  if (handlers.length > 1) {
-    handlers = ArrayPrototypeSlice(targetListeners[type]);
-  }
+  const handlers = ArrayPrototypeSlice(targetListeners[type]);
```

Comment:
I believe this was a useful optimization for us to keep our websocket perf up. Rather than reverting this and always cloning the array, I would suggest that we memoize the `handlers.length` instead so that in the case where we're not slicing the array, we'll only ever go through the loop once.

Reply:
Github won't allow me to added suggestions on deleted lines, but I think it would look like the code below:

```
  let handlers = targetListeners[type];
  const handlersLength = handlers.length;
  // Copy event listeners before iterating since the list can be modified during the iteration.
  if (handlersLength > 1) {
    handlers = ArrayPrototypeSlice(targetListeners[type]);
  }
  
for (let i = 0; i < handlersLength; i++) {
```

Reply:
Sounds good! Your suggestion has been committed 🙂